File: Data Structures & Algorithms/search-in-rotated-sorted-array-ii/submission-1.java
class Solution {
    public boolean search(int[] nums, int target) {
        int start = nums[0];
        int n = nums.length;
        int end = nums[n-1];

        if (target>=start){
            int j = 0;
            while(j<n && nums[j]<=target && nums[j]>=start){
                if(nums[j]==target){
                    return true;
                }
                j++;
            }
            return false;
        }
        else{
            int i = n-1;
            while(i>=0 && nums[i]>=target && nums[i]<=end ){
                if(nums[i]==target){
                    return true;
                }
                i--;
            }
            return false;
        }
    }
}

File: Data Structures & Algorithms/partition-equal-subset-sum/submission-1.py
class Solution:
    def canPartition(self, nums: List[int]) -> bool:
        n = len(nums)
        s = sum(nums)
        if s%2==1:
            return False
        target = s/2
        def dfs(i,currsum):
            if i>=n or currsum>target:
                return False
            elif currsum==target:
                return True
            else:
                return dfs(i+1,currsum+nums[i]) or dfs(i+1,currsum)
        return dfs(0,0)
